### PR TITLE
Feat: `EIP-7610` - non-empty storage create collision and upgrade evm-tests

### PR DESCRIPTION
**Pull request by mrLSD**
## Description

`EIP-7610` is:
> This EIP causes contract creation to throw an error when attempted at an address with pre-existing storage.

### New features

➡️  Implemented [ EIP-7610: Revert creation in case of non-empty storage](https://eips.ethereum.org/EIPS/eip-7610)
➡️  Refactored `evm-test` to [ethereum/tests v13.3](https://github.com/ethereum/tests)
➡️   Generalized cargo crates config fields

### Breaking changes

➡️  According to `EIP-7610`:
> This is an execution layer upgrade, and so it requires a hard fork.

➡️  `evm-tests` backward incompatible with [ethereum/tests v13.2](https://github.com/ethereum/tests)

### CODE REVIEW

File: src/executor/stack/executor.rs [hunk 1]
```diff
@@ -839,18 +847,10 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
...
-			if self.nonce(address) > U256::zero() {
-				let _ = self.exit_substate(StackExitKind::Failed);
-				return Capture::Exit((ExitError::CreateCollision.into(), None, Vec::new()));
-			}
-
-			self.state.reset_storage(address);
+		// Check create collision: EIP-7610
+		if self.is_create_collision(address) {
+			let _ = self.exit_substate(StackExitKind::Failed);
+			return Capture::Exit((ExitError::CreateCollision.into(), None, Vec::new()));
```
Comment (aleksuss) [inline src/executor/stack/executor.rs:853]: Did you miss `self.state.reset_storage(address);` or it's not needed anymore?

File: runtime/src/handler.rs [hunk 2]
```diff
@@ -35,6 +35,8 @@ pub trait Handler {
 	fn code(&self, address: H160) -> Vec<u8>;
 	/// Get storage value of address at index.
 	fn storage(&self, address: H160, index: H256) -> H256;
+	/// Check is storage of address empty.
```
Comment (aleksuss) [inline runtime/src/handler.rs:38]:
```suggestion
	/// Check if the storage of the address is empty.
```

File: src/backend/mod.rs [hunk 3]
```diff
@@ -82,6 +82,8 @@ pub trait Backend {
 	fn code(&self, address: H160) -> Vec<u8>;
 	/// Get storage value of address at index.
 	fn storage(&self, address: H160, index: H256) -> H256;
+	/// Check is storage of address empty.
```
Comment (aleksuss) [inline src/backend/mod.rs:85]:
```suggestion
	/// Check if the storage of the address is empty.
```

File: src/executor/stack/executor.rs [hunk 4]
```diff
@@ -703,6 +703,14 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		self.state.basic(address).nonce
 	}
 
+	/// Check is an existing account the "create collision".
```
Comment (aleksuss) [inline src/executor/stack/executor.rs:706]:
```suggestion
	/// Check if the existing account is "create collision".
```

File: src/executor/stack/executor.rs (hunk 1 above)
Comment (mrLSD) [reply src/executor/stack/executor.rs:853]: It checks storage is empty. We shouldn't touch storage data again, as it's empty anyway.

File: evm-tests/jsontests/Cargo.toml [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
-version = "0.42.0"
+version.workspace = true
 license = "GPL-3.0"
-authors = ["Aurora Labs <hello@aurora.dev>", "Wei Tang <hi@that.world>", "Parity Technologies <admin@parity.io>"]
-description = "SputnikVM - a Portable Blockchain Virtual Machine"
-repository = "https://github.com/sorpaas/rust-evm"
-keywords = ["no_std", "ethereum"]
-edition = "2021"
+authors.workspace = true
+description = "EVM jso tests"
```
Comment (birchmd) [inline evm-tests/jsontests/Cargo.toml:6]:
```suggestion
description = "EVM json tests"
```

**Review by birchmd — APPROVED** [review]

File: src/executor/stack/executor.rs (hunk 4 above)
Comment (mrLSD) [reply src/executor/stack/executor.rs]: Chenges: 74af9b66eb9506602602a1fca039bd86fc9480b4

File: src/backend/mod.rs (hunk 3 above)
Comment (mrLSD) [reply src/backend/mod.rs]: Chenges: 74af9b66eb9506602602a1fca039bd86fc9480b4

File: runtime/src/handler.rs (hunk 2 above)
Comment (mrLSD) [reply runtime/src/handler.rs]: Chenges: 74af9b66eb9506602602a1fca039bd86fc9480b4

**Review by aleksuss — APPROVED** [review]

**Pull request merged**